### DISCUSSION TITLE
Distributing fonts

### DISCUSSION THREAD

**Discussion by jamessouth** [fontforge/fontforge#5182] [Q&A] [answered]
When I generated my `.ttf` files, they were all about 2-3K on my system and on GitHub. When I `curl`ed them back to simulate a user installing them, they were 140K. I have not been able to find an explanation. I also noticed a Google Font was 90K on GitHub but 140K when downloaded. So then I thought I could subset my font, it only has 10 glyphs, how could it be 140K? When you open the `.ttf` in FontForge, there's a box for each Unicode code point. Sub-setting means cutting all that out and then the file size will be small, right? No. I ran it through [fonttools](https://github.com/fonttools/fonttools), it would cut the size down a bit to like 1K, but the downloaded size was still around 140K and all the code points were still there.

#### Observations:
The compressed file is what is used by a user's system.
A font's uncompressed size will be around 140K.
To download a compressed file from GitHub, you can use `curl` with the `-JOL` flags and get the raw data by appending the `?raw=true` query string to the url. 
Sub-setting means cutting out existing glyphs from a font. It does *not* mean removing unused Unicode code points such that their boxes don't show up in FontForge. They can be hidden with Encoding > Compact, but not eliminated.

Is that right? What is the compression being used? 

Thank you!

**Comment by ctrlcctrlv** [comment, marked as answer]
Sorry your question makes no sense. Encoding→Compact is a UI thing only. If you "curled it back" and got a larger file then something's wrong with either your conception of how that tool works or something other than FontForge.

**Reply by jamessouth** [reply]
ok I see now, thank you!

**Reply by jamessouth** [reply]
when you don't curl for the raw data, you get the whole web page. sorry! 🤦
